### PR TITLE
Allow to override the read_only option of a Smart Field

### PR DESCRIPTION
**Pull request by SeyZ**

### CODE REVIEW

File: lib/forest_liana/collection.rb [hunk 1]
```diff
@@ -14,16 +14,19 @@ def collection(collection_name, opts = {})
...
     def field(name, opts, &block)
+      opts[:read_only] = true unless opts.has_key?(:read_only)
+      opts[:is_searchable] = false unless opts.has_key?(:is_searchable)
+
       model.fields << opts.merge({
         field: name,
-        :'is-read-only' => true,
-        :'is-searchable' => false,
+        :'is-read-only' => opts[:read_only],
+        :'is-searchable' => opts['is_searchable'],
```
Comment (arnaudbesnier) [inline lib/forest_liana/collection.rb]: Will it activate the filters on such a smart field if `is_searchable` is set to `true`?

File: lib/forest_liana/collection.rb [hunk 2]
```diff
@@ -14,16 +14,19 @@ def collection(collection_name, opts = {})
...
 
     def field(name, opts, &block)
+      opts[:read_only] = true unless opts.has_key?(:read_only)
+      opts[:is_searchable] = false unless opts.has_key?(:is_searchable)
+
       model.fields << opts.merge({
         field: name,
-        :'is-read-only' => true,
-        :'is-searchable' => false,
+        :'is-read-only' => opts[:read_only],
```
Comment (arnaudbesnier) [inline lib/forest_liana/collection.rb]: I guess you have to implement a `setter` method to activate the write mode. The API should set the property to `false` automatically if the setter is defined. What do you think?

Comment (SeyZ) [inline lib/forest_liana/collection.rb]: Not sure to understand, here's how to set the readonly option on a smart field

```
field :fullname, type: 'String', read_only: false do
  # ...
end
```

Is it ok?

File: lib/forest_liana/collection.rb (hunk 1 above)
Comment (SeyZ) [inline lib/forest_liana/collection.rb]: yes

**Pull request merged**